interfaces/builtin/userns.go

```diff
@@ -0,0 +1,104 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	apparmor_sandbox "github.com/snapcore/snapd/sandbox/apparmor"
+	"github.com/snapcore/snapd/strutil"
+)
+
+const userNSSummary = `allows the ability to use user namespaces`
+
+// This interface is super-privileged
+const userNSBaseDeclarationSlots = `
+  userns:
+    allow-installation: false
```

Comment:
actually this doesn't look correct as it's about the installation of the slot which will be on the system snap, see snapd_control.go for something similar. allow-installation is not inherited from slot to plug

Reply:
hmm so I modelled this on the `lxd` interface and others like `docker` which use this similar construct - I guess I am confused since we seem to have two different ways to declare a plug as super-privileged - the docker way and the snapd-control way - would you be able to explain the difference to me?

Reply:
lxd and docker are app provided slots so it's the slot as shipped by their respective application snaps. here instead we have a system provided slot, in fact I think this interface is also missing setting:

                implicitOnCore:        true,
                implicitOnClassic:     true,

?

Reply:
ah ok - thanks for the clarification (apologies, I should have been able to figure that out myself). Fixed in  https://github.com/snapcore/snapd/pull/12530/commits/8855581b23efbdd8e8b97e3ec6bc6c840c82a3fc